Buddy/View/UserSettings.cs

```diff
@@ -0,0 +1,204 @@
+                        _postRepository.Update(post.Id, post);
+                    }
+                }
+                //pakeiciam username postRecord
+                foreach (var postRecord in postRecords)
+                {
+                    if (postRecord.Username == CurrentUser.UserName)
+                    {
+                        postRecord.Username = metrotextboxUsername.Text;
+                        _postRecordRepository.Update(CurrentUser.UserName, postRecord);
+                    }
+                }
+            }
+
+            if (ProfilePictureChanged) { CurrentUser.ProfilePicture = ImageToByteArray(pictureboxProfilePicture.Image); }
+
+            User updated = new User
+            {
+                Username = metrotextboxUsername.Text,
+                Email = metrotextboxEmail.Text,
+                Password = _userRepository.GetUserByUsername(CurrentUser.UserName).Password,
+                Faculty = metrocomboboxFaculty.Text,
+                StudyProgram = metrocomboboxStudyProgram.Text,
+                Semester = metrocomboboxSemester.Text,
+                ProfilePicture = CurrentUser.ProfilePicture
+            };
+
+            _userRepository.Update(id: CurrentUser.Id, item: updated);
+            CurrentUser.UserName = metrotextboxUsername.Text;
+            //reikes dar perdaryt kad nesaugoti visko
+            CurrentUser.Email = metrotextboxEmail.Text;
+            CurrentUser.Faculty = metrocomboboxFaculty.Text;
+            CurrentUser.StudyProgram = metrocomboboxStudyProgram.Text;
+            CurrentUser.Semester = metrocomboboxSemester.Text;
+
+            MessageBox.Show(this, "Changed succesfuly!");
+            this.Close();
+
+            var userProfile = new UserProfile(
+                    _postRepository,
+                    _postManager,
+                    _userRepository,
+                    _postRecordRepository,
+                    _postRecordManager
+                );
+            userProfile.Show();
+        }
+
+        private void OnFacultyChanged(object sender, EventArgs e)
+        {
+            metrocomboboxStudyProgram.Items.Clear();
+            metrocomboboxStudyProgram.ResetText();
+            metrocomboboxStudyProgram.Items.AddRange(FacultyReader.GetProgramNames(metrocomboboxFaculty.Text).ToArray());
+        }
+
+        private void OnButtonCancelClick(object sender, EventArgs e)
+        {
+           Close();
+
+           var userProfile = new UserProfile(
+                    _postRepository,
+                    _postManager,
+                    _userRepository,
+                    _postRecordRepository,
+                    _postRecordManager
+                );
+            userProfile.Show();
+        }
+
+        private void OnLinkChangePasswordClick(object sender, EventArgs e)
+        {
+            ChangePassword changePassword = new ChangePassword(_userRepository);
+            changePassword.Show();
```

Comment:
I feel it makes more sence, if user can have only one "Change Password" prompt open at a time.
```suggestion
            new ChangePassword(_userRepository).showDialog(this);
```